### ISSUE TITLE
If pinging with reconnect on and the connection goes down, infinite connect/reconnect loop

### ISSUE THREAD

**Issue by fhriley** [websocket-client/websocket-client#891]
Running this code:

```python
websocket.enableTrace(True)
while True:
    try:
        wsapp.run_forever(ping_interval=5, ping_timeout=3, reconnect=1)
    except Exception:
        log.exception('wsapp.run_forever() failed')
    finally:
        log.debug('ws thread exiting')
```

If the connection gets cut, the `send_ping` loop [will exit](https://github.com/websocket-client/websocket-client/blob/601a9022d6044bf6d414fdb1122916b65dc58ca7/websocket/_app.py#L254). Then when the connection returns, websocket-client will reconnect. But then `check()` will throw a `WebSocketTimeoutException` because `self.last_ping_tm` is [still set](https://github.com/websocket-client/websocket-client/blob/601a9022d6044bf6d414fdb1122916b65dc58ca7/websocket/_app.py#L429). `send_ping` is no longer running so it won't ever get updated and now you're stuck in an infinite connect/throw/reconnect loop.

**Comment by bubbleboy14** [comment]
Hey @fhriley, does https://github.com/websocket-client/websocket-client/pull/859 help?

**Comment by fhriley** [comment]
It seems like it should.

**Comment by engn33r** [comment]
The issue should be resolved with version 1.5.0, which includes a fix from PR #862 - the change made in #859 was included in #862
